### PR TITLE
feat: improve Sentry logging

### PR DESCRIPTION
**Pull request by bolinocroustibat**
Closes https://github.com/datagouv/data.gouv.fr/issues/1389:

- Update Sentry module and fix deps conflicts (see note)
- Send environment (using `SITE_ID`) and release info to Sentry
- Send profiling and traces to Sentry for performance analyzing
- Add an app-scoped config var `SENTRY_SAMPLE_RATE`

**Note:**
This PR adds `sentry-sdk[flask]` as an install dependency. I'm not sure why it wasn't there before - let me know if this is a mistake!
So since `sentry-sdk[flask]` depends on `urllib3` and `urllib3` depends on `requests` which were incompatible with newer versions of `sentry-sdk`, both pinned versions of `urllib3` and `requests` have been updated in this PR.

### CODE REVIEW

File: udata/sentry.py [hunk 1]
```diff
@@ -56,10 +62,24 @@ def init_app(app):
...
+            release=f"udata@{package_version('udata')}",
+            environment=app.config.get('SITE_ID', None),
+            # Set traces_sample_rate to 1.0 to capture 100%
+            # of transactions for performance monitoring.
+            # Sentry recommends adjusting this value in production.
+            traces_sample_rate=sentry_sample_rate,
+            # Experimental profiling
+            _experiments={
+                'profiles_sample_rate': sentry_sample_rate,
+            },
```
Comment (maudetes) [inline udata/sentry.py:82]: I would say that we should log 100% of errors by default for now (and make it configurable) but only a portion of profiling indeed. Ignore this comment if this is what it does ;)

**Review by maudetes — APPROVED** [review]
Thank you! We can probably test it on our dev envs :)

Don't forget to add a changelog entry ;)

File: udata/sentry.py (hunk 1 above)
Comment (bolinocroustibat) [reply udata/sentry.py:82]: Yes after some inquiry, this is only for perf so let's keep as it is.

Comment (bolinocroustibat) [reply udata/sentry.py]: But: Sentry sample rate has been changed as an app config.

**Comment by bolinocroustibat** [comment]
> Thank you! We can probably test it on our dev envs :)
> 
> Don't forget to add a changelog entry ;)

Indeed. Added a changelog entry.

File: requirements/install.pip [hunk 2]
```diff
@@ -279,7 +281,7 @@ uritools==4.0.3
     # via urlextract
 urlextract==0.14.0
     # via -r requirements/install.in
-urllib3==1.25.11
+urllib3==1.26.19
```
Comment (maudetes) [inline requirements/install.pip:284]: I think you should compute pip-compile on all impacted dep files.
It should have gotten triggered by the pip-tool pre-commit hook.
I've updated its version in https://github.com/opendatateam/udata/pull/3086/commits/5982e2a8f5ee0f24ac779887870f2415527bd9cc.

**Review by magopian — APPROVED** [review]
LGTM, nice work 👍

**Pull request merged**